content/chapters/software-stack/lab/support/app-investigate/get_app_types.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+find /usr/bin /bin /usr/sbin /sbin -type f -exec file {} \; |
+    awk '
+BEGIN {
+    num_bin = 0;
+    num_perl = 0;
+    num_shell = 0;
+    num_python = 0;
+    num_others = 0;
+}
+
+/.*shell script.*/ {
```

Comment:
There should be no need to match the entire line if neither `^` or `$` are used. The regexes can be simplified as `/shell script/` (or equivalent for each match), since matches are by default performed on any part of a string.